Source/WebCore/css/CSSGridLineValue.cpp

```diff
@@ -63,22 +63,22 @@ Ref<CSSGridLineValue> CSSGridLineValue::create(RefPtr<CSSPrimitiveValue>&& spanV
 
 bool CSSGridLineValue::equals(const CSSGridLineValue& other) const
 {
-    if (m_spanValue) {
-        if (!other.m_spanValue || !m_spanValue->equals(*other.m_spanValue))
+    if (RefPtr spanValue = m_spanValue, otherSpanValue = other.m_spanValue; spanValue) {
+        if (!otherSpanValue || !spanValue->equals(*otherSpanValue))
             return false;
-    } else if (other.m_spanValue)
+    } else if (otherSpanValue)
         return false;
```

Comment:
I'd probably factor this repeated logic into a lambda:

```
auto equals = [](CSSPrimitiveValue* value, CSSPrimitiveValue* otherValue) {
    if ((!value && otherValue) || (value && !otherValue))
        return false;
    return (!value && !otherValue) || value->equals(otherValue);
};
```
then use it like this:
```
if (!equals(protectedSpanValue(), other.protectedSpanValue())
    return false;
if (!equals(protectedNumericValue(), protectedNumericValue())
    return false;
if (!equals(protectedGridLineName(), protectedGridLineName())
    return false;
```